torch_xla/csrc/aten_cpu_fallback.cpp

```diff
@@ -0,0 +1,48 @@
+#include "torch_xla/csrc/aten_cpu_fallback.h"
+
+#include <tensorflow/compiler/xla/xla_client/debug_macros.h>
+#include <tensorflow/compiler/xla/xla_client/metrics.h>
+#include <tensorflow/compiler/xla/xla_client/tf_logging.h>
+#include <torch_xla/csrc/function_call_tracker.h>
+
+#include <unordered_map>
+
+namespace torch_xla {
+
+std::unordered_map<std::string, ::xla::metrics::Counter*>
+    _cpu_fallback_counters;
+
+void xla_cpu_fallback(const c10::OperatorHandle& op, torch::jit::Stack* stack) {
+  XLA_FN_TRACK(3);
+  const auto name = c10::toString(op.operator_name());
+
+  // Manually applying the XLA_COUNTER macro.
```

Comment:
So, after staring at some failing C++ tests for a while, I finally realized that some tests were failing because of my usage of the `XLA_COUNTER` macro here :(

It's defined in the XLA repo [here](https://github.com/pytorch/xla/blob/35625e78551b16c4d29d3968329fd7d4f1cb53fc/third_party/xla_client/metrics.h#L172), and defined a local static counter that's unique to the name that you pass in. Which means that it silently does bad things if you try to call the macro with different operator names from the same piece of source code. If I call `xla_cpu_fallback()` once with `add`, then with `mul`, the counter for `add` will get incremented twice.

Maybe it's worth a patch to xla to add a different version of the macro? For now, I just hardcoded what the macro does here, with a global mapping of counters for every op that the CPU fallback is called with.

Reply:
nice catch! `XLA_COUNTER` is designed for being used with different names, I think it is fine to leave it as it is.